src/NLog/LayoutRenderers/Wrappers/CachedLayoutRendererWrapper.cs

```diff
@@ -47,7 +48,21 @@ namespace NLog.LayoutRenderers.Wrappers
     [ThreadAgnostic]
     public sealed class CachedLayoutRendererWrapper : WrapperLayoutRendererBase
     {
-        private string cachedValue;
+        /// <summary>
+        /// A value indicating when the cache is cleared.
+        /// </summary>
+        [Flags]
+        public enum ClearCacheOption 
+        { 
+            /// <summary>Never clear the cache.</summary>
+            None = 0,
```

Comment:
I think 'None' is a bit tricky, because what will this do? ` var option = ClearCacheOption.None | ClearCacheOption .OnClose` ?